packages/sandbox/src/cloudwatch_logs_monitor.ts

```diff
@@ -0,0 +1,284 @@
+   */
+  constructor(readonly cloudWatchLogsClient: CloudWatchLogsClient) {
+    this.startTime = Date.now();
+  }
+
+  /**
+   * resume writing/printing events
+   */
+  activate = (): void => {
+    this.active = true;
+    this.scheduleNextTick(0);
+  };
+
+  /**
+   * deactivates the monitor so no new events are read
+   * use case for this is when we are in the middle of performing a deployment
+   * and don't want to interweave all the logs together with the CFN
+   * deployment logs
+   *
+   * Also resets the start time to be when the new deployment was triggered
+   * and clears the list of tracked log groups
+   */
+  deactivate = (): void => {
+    this.active = false;
+    this.startTime = Date.now();
+    this.allLogGroups.splice(0, this.allLogGroups.length);
+  };
+
+  /**
+   * Adds CloudWatch log groups to read log events from.
+   * Since we could be watching multiple logs groups we need a friendly
+   * name for to associate the log group to make it easier
+   * for the user to identify which log groups are being monitored
+   * @param friendlyResourceName The friendly name of the resource that is being monitored
+   * @param logGroupName The log group to read events from
+   */
+  addLogGroups = (friendlyResourceName: string, logGroupName: string): void => {
+    this.allLogGroups.push({
+      logGroupName,
+      startTime: this.startTime,
+    });
+    this.logGroupEventDisplay[logGroupName] = {
+      friendlyName: friendlyResourceName,
+      color: this.getNextColorForLogGroup(),
+    };
+  };
+
+  /**
+   * If output location file is specified, the logs will be appended to that file.
+   * If the file doesn't exist it will be created.
+   * @param outputLocation file location
+   */
+  setOutputLocation = (outputLocation: string) => {
+    if (!outputLocation) {
+      return;
+    }
+    const targetPath = path.isAbsolute(outputLocation)
+      ? outputLocation
+      : path.resolve(process.cwd(), outputLocation);
+    this.printer = new Printer(
+      LogLevel.INFO,
+      fs.createWriteStream(targetPath, { flags: 'a', autoClose: true })
+    );
+  };
+
+  /**
+   * Pick the next color in the object `colors` in round robin fashion
+   */
+  private getNextColorForLogGroup = () => {
+    return colorNames[
+      this.allLogGroups.length % colorNames.length
+    ] as ColorName;
+  };
+
+  private scheduleNextTick = (sleep: number): void => {
+    setTimeout(() => void this.tick(), sleep);
+  };
+
+  private tick = async (): Promise<void> => {
+    if (!this.active) {
+      return;
+    }
+    try {
+      const events = await this.readNewEvents();
+      events.forEach((event: CloudWatchLogEvent) => {
+        this.print(event);
+      });
+    } catch (e) {
+      throw new AmplifyFault(
+        'CloudWatchStreamingFault',
+        { message: `Error occurred while monitoring logs: %s` },
+        e instanceof Error ? e : undefined
+      );
+    }
+
+    this.scheduleNextTick(SLEEP_MS);
```

Comment:
Is there any reason why we didn't try `setTimeout` or `setInterval` ? Or is this just equivalent?

Reply:
it's just a wrapper over `setTimeout`